Source/ARTJsonLikeEncoder.m

```diff
@@ -128,7 +128,7 @@ - (NSData *)encodeTokenDetails:(ARTTokenDetails *)tokenDetails error:(NSError **
     return [self encode:[self tokenDetailsToDictionary:tokenDetails] error:error];
 }
 
-- (NSData *)encodeDeviceDetails:(ARTDeviceDetails *)deviceDetails error:(NSError **)error {
+- (NSData *)encodeDeviceDetails:(ARTLocalDevice *)deviceDetails error:(NSError **)error {
```

Comment:
```suggestion
- (NSData *)encodeDeviceDetails:(ARTLocalDevice *const)deviceDetails error:(NSError **)error {
```

I think it's also possible to make the pointer to the pointer `const` also, but I forget the syntax (am reviewing in GitHub UI without Xcode open).